app/views/ThemeView.tsx

```diff
@@ -92,45 +90,48 @@ const ThemeView = ({ navigation }: IThemeViewProps): React.ReactElement => {
 			logEvent(events.THEME_SET_DARK_LEVEL, { dark_level: value });
 			changes = { darkLevel: value as TDarkLevel };
 		}
-		_setTheme(changes);
+		handleTheme(changes);
 	};
 
-	const _setTheme = (theme: Partial<IThemePreference>) => {
+	const handleTheme = (theme: Partial<IThemePreference>) => {
 		const newTheme: IThemePreference = { ...(themePreferences as IThemePreference), ...theme };
 		if (setTheme) {
 			setTheme(newTheme);
 			UserPreferences.setMap(THEME_PREFERENCES_KEY, newTheme);
 		}
 	};
 
-	const renderIcon = () => <List.Icon name='check' color={themes[theme].tintColor} />;
-
-	const renderItem = ({ item }: { item: ITheme }) => {
-		const { label, value } = item;
-		return (
-			<>
-				<List.Item
-					title={label}
-					onPress={() => onClick(item)}
-					testID={`theme-view-${value}`}
-					right={() => (isSelected(item) ? renderIcon() : null)}
-				/>
-				<List.Separator />
-			</>
-		);
-	};
+	const RenderItem = React.memo(({ theme }: { theme: ITheme }) => (
```

Comment:
This `React.memo` doesn't work, because the functions `onClick` and `isSelected` are recreated all re-render. In this case, we should use the `useCallback`.

What do you think about removing the RenderItem from inside the `father` and create a new component to `RenderItem`.

Reply:
I prefer useCallback

Reply:
Separate RenderItem to another file `Item`